lib/rubocop/cop/style/method_call_with_args_parentheses/omit_parentheses.rb

```diff
@@ -202,6 +203,17 @@ def assigned_before?(node, target)
           def inside_string_interpolation?(node)
             node.ancestors.drop_while { |a| !a.begin_type? }.any?(&:dstr_type?)
           end
+
+          def assignment_in_condition?(node)
```

Comment:
I think it'd be better to implement this in terms of pattern matching, as logic that goes through 2 levels of parents is both brittle and hard to understand.

Reply:
I cannot use `def_node_pattern` here because of the way this cop is organized. This is a module that is mixed in into `Style/MethodCallWithArgsParentheses`. I can invoke `NodePattern` by itself, but I am yet to find a good pattern that looks for a grandparent.

I did had a redundant grandparent check, though. The new implementation is simpler.